src/main/kotlin/org/wycliffeassociates/otter/common/domain/resourcecontainer/project/markdown/MarkdownProjectReader.kt

```diff
@@ -1,13 +1,110 @@
 package org.wycliffeassociates.otter.common.domain.resourcecontainer.project.markdown
 
+import org.wycliffeassociates.otter.common.collections.tree.OtterTree
+import org.wycliffeassociates.otter.common.collections.tree.OtterTreeNode
 import org.wycliffeassociates.otter.common.collections.tree.Tree
+import org.wycliffeassociates.otter.common.collections.tree.TreeNode
+import org.wycliffeassociates.otter.common.data.model.Collection
+import org.wycliffeassociates.otter.common.data.model.Content
 import org.wycliffeassociates.otter.common.domain.resourcecontainer.ImportResult
 import org.wycliffeassociates.otter.common.domain.resourcecontainer.project.IProjectReader
 import org.wycliffeassociates.resourcecontainer.ResourceContainer
 import org.wycliffeassociates.resourcecontainer.entity.Project
+import java.io.BufferedReader
+import java.io.File
+import java.util.*
+
+private const val FORMAT = "text/markdown"
+private val extensions = Regex(".+\\.(md|mkdn?|mdown|markdown)$", RegexOption.IGNORE_CASE)
 
 class MarkdownProjectReader: IProjectReader {
-    override fun constructProjectTree(container: ResourceContainer, project: Project): Pair<ImportResult, Tree> {
-        TODO("The MarkdownProjectReader isn't yet implemented.")
+    override fun constructProjectTree(container: ResourceContainer, project: Project)
+            : Pair<ImportResult, Tree> {
+        val projectRoot = container.dir.resolve(project.path)
+        return projectRoot
+                .buildFileTree()
+                .filterMarkdownFiles()
+                ?.map<Any> { f -> contentList(f) ?: collection(f, projectRoot) }
+                ?.flattenContent()
+                ?.let { Pair(ImportResult.SUCCESS, it) }
+                ?: Pair(ImportResult.LOAD_RC_ERROR, Tree(Unit))
+    }
+
+    private fun fileToId(f: File): Int =
+        f.nameWithoutExtension.toIntOrNull() ?: 0
+
+    private fun fileToSlug(f: File, root: File): String =
+        f.toRelativeString(root.parentFile)
+                .split('/', '\\')
+                .map { it.toIntOrNull()?.toString() ?: it }
+                .joinToString("_")
+
+    private fun bufferedReaderProvider(f: File): (() -> BufferedReader)? =
+            if (f.isFile) { { f.bufferedReader() } } else null
+
+    private fun collection(f: File, projectRoot: File): Collection =
+            collection(fileToSlug(f, projectRoot), fileToId(f))
+
+    private fun collection(slug: String, id: Int): Collection =
+            Collection(
+                    sort = id,
+                    slug = slug,
+                    labelKey = "helpcollection",
```

Comment:
this sounds like it would be too vague of a key?

Reply:
True.  Two answers though:
1. The specific (shortsighted) answer: It doesn't matter, because these collections aren't written to the DB in any of our use-cases; the content gets attached to the existing collection hierarchy from the already-imported Bible RC.
2. The general answer: This MarkdownProjectReader should not limit itself to existing use-cases, and so I should either parameterize this value, or use something from the manifest instead of this hardcoded value.
I'll push an update.

Reply:
Done.  We'll eventually want this key to be more configurable (like when we start reading Bibles in Markdown format), and MarkdownProjectReader should be parametrized with a function to create the key from the RC+slug (or something) but I think we should defer that impl until we know more about what we'll need.  For now, it just uses `dublinCore::identifier`.